src/color.lisp

```diff
@@ -36,13 +36,20 @@
 (defun make-color (r g b &optional (a 255))
   (make-instance 'color :r r :g g :b b :a a))
 
-(defun copy-color (color)
-  (if (typep color 'rl-color)
-      (make-color (r color) (g color) (b color) (a color))
-      (make-color (claylib/ll:color.r color)
-                  (claylib/ll:color.g color)
-                  (claylib/ll:color.b color)
-                  (claylib/ll:color.a color))))
+(defun copy-color (color &optional into)
+  "Copy a given COLOR by allocating a new one unless an existing rl-color, INTO, is given."
+  (if into
+      ;; TODO handle case where color is not an rl-color (∴ cannot use rgba accessors)
+      (setf (r into) (r color)
+            (g into) (g color)
+            (b into) (b color)
+            (a into) (a color))
+      (if (typep color 'rl-color)
+          (make-color (r color) (g color) (b color) (a color))
+          (make-color (claylib/ll:color.r color)
+                      (claylib/ll:color.g color)
+                      (claylib/ll:color.b color)
+                      (claylib/ll:color.a color)))))
```

Comment:
This change is triggering my spidey senses. Does the music stream example (which does not use `into`) work on its own if you have not run the other one?

Reply:
No, I'm getting the same behaviour with fresh lisps for each. I was also getting that behaviour before implementing this change. This change, by the way, is for circumstances like in `module-playing.lisp` where you want to set an existing colour to another without consing. I'm sure you gathered that, I'm just defending its existence (not its implementation).